addon/components/ember-table.js

```diff
@@ -0,0 +1,564 @@
+  // more content than they can display.
+  // TODO: Currently must be kept in sync with the LESS file.
+  minHeaderHeight: 30,
+
+  // The footer height in pixels.
+  // TODO: Currently must be kept in sync with the LESS file.
+  footerHeight: 30,
+
+  // Enables or disables the header block.
+  hasHeader: true,
+
+  // Enables or disables the footer block.
+  // TODO(new-api): Control this via `numFooterRows` and remove from API
+  hasFooter: true,
+
+  enableColumnReorder: true,
+
+  // Allow users to select the content of table cells.
+  enableContentSelection: false,
+
+  // Sets which column resizing behavior to use. Possible values are
+  // <code>'standard'</code> (resizing a column pushes or pulls all other
+  // columns) and <code>'fluid'</code> (resizing a column steals width from
+  // neighboring columns).
+  columnMode: 'standard',
+
+  // Sets which row selection behavior to follow. Possible values are 'none'
+  // (clicking on a row does nothing), 'single' (clicking on a row selects it
+  // and deselects other rows), and 'multiple' (multiple rows can be selected
+  // through ctrl/cmd-click or shift-click).
+  selectionMode: 'single',
+
+  // ---------------------------------------------------------------------------
+  // API - Outputs
+  // ---------------------------------------------------------------------------
+
+  // An array of the rows currently selected. If `selectionMode` is set to
+  // 'single', the array will contain either one or zero elements.
+  selection: Ember.computed(function(key, val) {
+    var selectionMode = this.get('selectionMode');
+    if (arguments.length > 1 && val) {
+      this.get('persistedSelection').clear();
+      this.get('rangeSelection').clear();
+      switch (selectionMode) {
+        case 'single':
+          this.get('persistedSelection').addObject(val);
+          break;
+        case 'multiple':
+          this.get('persistedSelection').addObjects(val);
+      }
+    }
+    var selection = this.get('persistedSelection').copy().addObjects(this.get('rangeSelection'));
+    switch (selectionMode) {
+      case 'none':
+        return null;
+      case 'single':
+        return selection[0] || null;
+      case 'multiple':
+        return selection;
+    }
+  }).property('persistedSelection.[]', 'rangeSelection.[]', 'selectionMode'),
+
+  // ---------------------------------------------------------------------------
+  // Internal properties
+  // ---------------------------------------------------------------------------
+
+  // Special flag used by child views to look up this component using
+  // nearestWithProperty()
+  isEmberTable: true,
+
+  columnsFillTable: true,
+
+  init: function() {
+    this._super();
+    if (!Ember.$.ui) {
+      throw 'Missing dependency: jquery-ui';
+    }
+    if (!Ember.$().mousewheel) {
+      throw 'Missing dependency: jquery-mousewheel';
+    }
+    if (!Ember.$().antiscroll) {
```

Comment:
This shouldn't be needed in Ember CLI, since it's much harder to misconfigure when the install pulls in dependencies on its own. But it's still useful for the globals version, so I'll leave it for now.